packages/taxdat/R/setup_helpers.R

```diff
@@ -1,3 +1,164 @@
+#' @include config_helpers.R file_name_functions.R run_stan_helpers.R
+#' @title check_update_config
+#' @description Check whether parameter values exist or are valid and then update them 
+#' @param config_fname pathway to the config 
+#' @param covariate_list where to find the file that lists all the covariate names 
+#' @return the actual config file 
+#' @export
+check_update_config <- function(config_fname, covariate_list = NULL){
+  
+  ### Read in the config file first 
+  cholera_testing <- as.logical(Sys.getenv("CHOLERA_TESTING", "FALSE"))
+  config_file <- yaml::read_yaml(config_fname)
+
+  ### If no file exists to tell us what covariates exist, here's a hard coded list
+  if(is.null(covariate_list)){
+    all_covariates <- c("", "pop", "dist_to_water",              
+                        "dist_to_lakes", "dist_to_rivers",              
+                        "dist_to_coast", "san_access",                  
+                        "water_access", "open_defe",            
+                        "stunting", "wasting", 
+                        "access_cities", "access_healthcare_motorized", 
+                        "access_healthcare_walking")
+  }else{
+    all_covariates <- names(yaml::read_yaml(covariate_list)) #tmp default: "Layers/middle-distance_covariates.yml"
+  }
+
+  ### Make the check list 
+  check_list <- list(
+    name = "no-check", 
+    countries = "no-check", 
+    countries_name = "no-check", 
+    country_data_report_filename = function(nm, cty, ctynm, param){
+                if(is.null(param)){return(paste(nm, cty, ctynm, "country-data-report.html", sep="_"))}
+                return(param)
+              }, 
+    aoi = function(param){
+        if(cholera_testing){return(param)}
+        return("raw")
+      }, 
+    res_space = "no-check", 
+    res_time = as.function(check_time_res), 
+    grid_rand_effects_N = as.function(check_grid_rand_effects_N), 
+    case_definition = as.function(check_case_definition), 
+    start_time = function(param){
+                if(is.null(param)){stop("The start_time parameter should not be blank because there is no default")}
+                return(param)
+              },   
+    end_time = function(param){
+              if(is.null(param)){stop("The end_time parameter should not be blank because there is no default")}
+              return(param)
+            }, 
+    data_source = function(param){
+                if(is.null(param)){return("sql")}
+                return(param)
+              }, 
+    ovrt_metadata_table = function(param){
+                        if(is.null(param)){return("no")}
+                        return(param)
+                      }, 
+    OCs = "no-check", 
+    taxonomy = function(param){
```

Comment:
is taxonomy actually a required config parameter? I didn't think it was necessary to keep it...